libsolidity/formal/SMTChecker.cpp

```diff
@@ -1456,7 +1473,12 @@ TypePointer SMTChecker::typeWithoutPointer(TypePointer const& _type)
 void SMTChecker::mergeVariables(vector<VariableDeclaration const*> const& _variables, smt::Expression const& _condition, VariableIndices const& _indicesEndTrue, VariableIndices const& _indicesEndFalse)
 {
 	set<VariableDeclaration const*> uniqueVars(_variables.begin(), _variables.end());
-	for (auto const* decl: uniqueVars)
+	mergeVariables(uniqueVars, _condition, _indicesEndTrue, _indicesEndFalse);
+}
+
+void SMTChecker::mergeVariables(set<VariableDeclaration const*> const& _variables, smt::Expression const& _condition, VariableIndices const& _indicesEndTrue, VariableIndices const& _indicesEndFalse)
+{
+	for (auto const* decl: _variables)
```

Comment:
Can that lead to nondeterminism?

Reply:
What exactly, the order of traversal? Don't think so, their SSA index doesn't depend on that.

Reply:
What about the "SMT aux string"?

Reply:
Tracked in #6411